web/pages/Chat/ChatDetail.tsx

```diff
@@ -81,6 +81,16 @@ const ChatDetail: Component = () => {
 
     return !isMember
   })
+  
+  const waitingMsg = createMemo(() => {
+    if (!msgs.waiting) return
+
+    return emptyMsg({
+      charId: msgs.waiting?.mode !== 'self' ? chats.char?._id : undefined,
+      userId: msgs.waiting?.mode === 'self' ? msgs.waiting.userId || user.user?._id : undefined,
+      message: msgs.partial || '',
```

Comment:
I initially tried passing `msgs.partial` in the call to `emptyMsg` within the `<Show when={msgs.waiting}>` component, which was evidently not sufficient to cause the component to update, I assume because `msgs.waiting` itself was not changing.

No clue what the idiomatic way to do this in Solid is.

Reply:
if you want a value to depend on a signal's curre
 value, you must wrap that value in a function or a createMemo, so what you did looks fine